src/Dapr.Jobs/DaprJobsGrpcClient.cs

```diff
@@ -0,0 +1,236 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2024 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using System.Net.Http.Headers;
+using System.Reflection;
+using Dapr.Jobs.Models;
+using Dapr.Jobs.Models.Responses;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using Grpc.Core;
+using Autogenerated = Dapr.Client.Autogen.Grpc.v1;
+
+namespace Dapr.Jobs;
+
+/// <summary>
+/// A client for interacting with the Dapr endpoints.
+/// </summary>
+internal sealed class DaprJobsGrpcClient : DaprJobsClient
+{
+    /// <summary>
+    /// Present only for testing purposes.
+    /// </summary>
+    internal readonly HttpClient httpClient;
+    
+    /// <summary>
+    /// Used to populate options headers with API token value.
+    /// </summary>
+    internal readonly KeyValuePair<string, string>? apiTokenHeader;
+
+    private readonly Autogenerated.Dapr.DaprClient client;
+    private readonly string userAgent = UserAgent().ToString();
+
+    // property exposed for testing purposes
+    internal Autogenerated.Dapr.DaprClient Client => client;
+    
+    internal DaprJobsGrpcClient(
+        Autogenerated.Dapr.DaprClient innerClient,
+        HttpClient httpClient,
+        KeyValuePair<string, string>? apiTokenHeader)
+    {
+        this.client = innerClient;
+        this.httpClient = httpClient;
+        this.apiTokenHeader = apiTokenHeader;
+
+        this.httpClient.DefaultRequestHeaders.UserAgent.Add(UserAgent());
```

Comment:
Nit: can you pass `this.userAgent`?

Reply:
Looking at it, we don't even need that line. It's covered by the later invocation of `CreateCallOptions` to set up the various `CallOptions` on each request which in turn is using `this.userAgent`.